test/tests/val-2g/comparison_val-2g.py

```diff
@@ -0,0 +1,111 @@
+else:                                  # if in test folder
+    csv_folder = "./experimental_data.csv"
+experiment_data = pd.read_csv(csv_folder)
+experiment_time = experiment_data['time']
+experiment_temperature = experiment_data['temperature']
+experiment_flux_673 = experiment_data['Exp - 400C']
+experiment_flux_873 = experiment_data['Exp - 600C']
+experiment_flux_973 = experiment_data['Exp - 700C']
+
+# Setup mask using a random sampling method to reduce density of scatter plot points in experimental
+# data (there is a lot of "noise").
+# These preserve 10% of the data points. (adjusted within the 'size' parameter).
+mask = np.random.choice(len(experiment_flux_673), size=int(len(experiment_flux_673) * 0.1), replace=False)
+
+
+file_base = 'val-2g_comparison'
+############################ desorption flux atom/m$^2$/s ############################
+fig = plt.figure(figsize=[6.5, 5.5])
+gs = gridspec.GridSpec(1, 1)
+ax = fig.add_subplot(gs[0])
+
+# TMAP8
+ax.semilogy(simulation_time_673-12000, simulation_flux_left_673, linestyle='-', label=r"673 K", c='blue', linewidth=3.0)
+ax.semilogy(simulation_time_873-12000, simulation_flux_left_873, linestyle='-', label=r"873 K", c='red', linewidth=3.0)
+ax.semilogy(simulation_time_973-12000, simulation_flux_left_973, linestyle='-', label=r"973 K", c='green', linewidth=3.0)
+
+# TMAP4
+ax.semilogy(simulation_time_TMAP4, simulation_flux_TMAP4_673, linestyle='--', label=r"TMAP4 - 673 K", c='blue', linewidth=3.0, alpha=0.7)
+ax.semilogy(simulation_time_TMAP4, simulation_flux_TMAP4_873, linestyle='--', label=r"TMAP4 - 873 K", c='red', linewidth=3.0, alpha=0.7)
+ax.semilogy(simulation_time_TMAP4, simulation_flux_TMAP4_973, linestyle='--', label=r"TMAP4 - 973 K", c='green', linewidth=3.0, alpha=0.7)
+
+# Experimental Data
+s = (3.0)**2
+ax.scatter(experiment_time[mask], experiment_flux_673[mask], marker='o', s=s, label=r"Exp - 673 K", c='blue')
+ax.scatter(experiment_time[mask], experiment_flux_873[mask], marker='o', s=s, label=r"Exp - 873 K", c='red')
+ax.scatter(experiment_time[mask], experiment_flux_973[mask], marker='o', s=s, label=r"Exp - 973 K", c='green')
+
+
+# Font sizes for labels and axes
+SMALL_SIZE = 10
+MEDIUM_SIZE = 14
+BIGGER_SIZE = 16
+
+ax.set_xlabel(u'Time (s)', weight='bold', fontsize=BIGGER_SIZE)
+ax.set_ylabel(u"Desorption flux (H$_2$/m$^2$/s)", weight='bold', fontsize=BIGGER_SIZE)
+ax.legend(loc="upper left", fontsize=SMALL_SIZE)
+ax.set_ylim(bottom=6e15, top=4.0e18)
+ax.set_xlim(left=0,right=7200)
+ax.xaxis.set_ticks(np.arange(0, 7205, 1800))
+plt.grid(visible=True, which='major', color='0.65', linestyle='--', alpha=0.3)
+# tmap_flux_for_rmspe = numerical_solution_on_experiment_input(experiment_time, simulation_time_TMAP7, simulation_flux_left_TMAP7/2 + flux_environment)
+# RMSE = np.sqrt(np.mean((tmap_flux_for_rmspe-experiment_flux)**2) )
+# RMSPE = RMSE*100/np.mean(experiment_flux)
+# ax.text(6000.0,0.85e18, 'RMSPE = %.2f '%RMSPE+'%',fontweight='bold')
```

Comment:
Why is this commented out? Maybe it's just because this is still in WIP stage, but calculating the RMSPE between TMAP8's predictions and the experimental data is a good thing to add.